.github/workflows/main.yml

```diff
@@ -0,0 +1,51 @@
+name: Continuous integration
+
+on:
+  # Trigger the workflow on push or pull request,
+  push:
+  # Trigger the workflow via a manual run
+  workflow_dispatch:
+
+# Ensures that only one deploy task per branch/environment will run at a time.
+concurrency:
+  group: environment-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  install:
+    name: Install dependencies
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+      - name: Install dependencies
+        uses: ./.github/actions/install
+
+  unit-test:
+    name: Run unit test
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    needs: [install]
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+      - name: Install dependencies
+        uses: ./.github/actions/install
+      - name: Run tests
+        run: python3 -m unittest discover
+
+  lint:
```

Comment:
if we are going to swap out pylint for ruff, maybe we could do it in a separate commit? and maybe through Make (since that is what we are using to build, still)

Reply:
missed that we had pylint set up already. changing to Pylint instead :+1: